pkgs/development/compilers/idris2/default.nix

```diff
@@ -23,14 +32,10 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-6CTn8o5geWSesXO7vTrrV/2EOQ3f+nPQ2M5cem13ZSY=";
   };
 
-  # We do not add any propagatedNativeBuildInputs because we do not want the
-  # executables idris2 produces to depend on the nix-store. As such, it is left
-  # to the user to guarantee chez (or any other codgen dependency) is available
-  # in the path during compilation of programs with idris2.
   strictDeps = true;
-  nativeBuildInputs = [ makeWrapper clang chez ]
+  nativeBuildInputs = [ makeWrapper clang platformChez ]
     ++ lib.optional stdenv.isDarwin [ zsh ];
-  buildInputs = [ gmp ];
+  buildInputs = [ platformChez gmp ];
```

Comment:
Is it necessary to have platformChez both as a buildInput and a nativeBuildInput?

Reply:
I believe we need chez as both, a run-time dependency and as a build dependency. So I think, yes? This is also taken from the idris2 project nix derivation.

Reply:
I think buildInputs will be available both in build and runtime so I think removing it from nativeDependecies should still work.